Assets/Script/Misc/Types/Collections/SongCollection.cs

```diff
@@ -80,12 +81,12 @@ public void Reset()
             }
             sorted = origin;
         }
-        static SongDetail[] Sort(SongDetail[] origin,SortType sortType)
+        static SongDetail[] Sort(SongDetail[] origin, SortType sortType)
```

Comment:
static SongDetail[] Sort(SongDetail[] origin, SortType sortType)
{
    if (origin.IsEmpty())
        return origin;

    IEnumerable<SongDetail> result = sortType switch
    {
        SortType.ByTime => origin.OrderByDescending(o => o.AddTime),
        SortType.ByDiff => origin.OrderByDescending(o => o.Levels[4]),
        SortType.ByDes => origin.OrderBy(o => o.Designers[4]),
        SortType.ByTitle => origin.OrderBy(o => o.Title),
        _ => origin
    };

    return result.ToArray();  
}
Maybe a match expression would be cleaner here

Reply:
Adopted in commit 1a47ab518fb779e13cc63b4c49d0ea61770ed97c